src/item/HangingSign.php

```diff
@@ -40,13 +43,21 @@ public function __construct(
 		parent::__construct($identifier, $name);
 	}
 
-	public function getPlacementBlock(Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector) : Block{
+	public function getPlacementBlock(?Player $player, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector) : Block{
 		//we don't verify valid placement conditions here, only decide which block to return
-		$result = $face === Facing::DOWN ?
-			$blockReplace->getSide(Facing::UP)->getSupportType(Facing::DOWN) === SupportType::CENTER ?
+		if($face === Facing::DOWN){
+			if($player !== null && $player->isSneaking()){
+				return clone $this->centerPointCeilingVariant;
+			}
+
+			$support = $blockReplace->getSide(Facing::UP);
+			$result = (($support instanceof CeilingEdgesHangingSign || $support instanceof WallHangingSign) && ($player === null || Facing::axis($player->getHorizontalFacing()) !== Facing::axis($support->getFacing()))) ||
```

Comment:
I don't think this condition is quite right

In this code, center sign is placed if:

- Support is a ceiling edges sign or wall hanging sign, and
- Player is facing perpendicular to the sign

Or, support type is CENTER

However, a ceiling center sign can also be hung from another ceiling center sign, which this code doesn't account for (it'll select a ceiling edges sign).

The code is quite complex and hard to reason about, it could use some explanatory comments